api/config/v1alpha1/metric_types.go

```diff
@@ -29,6 +37,19 @@ type MetricSink struct {
 	OpenTelemetry *OpenTelemetrySink `json:"openTelemetry,omitempty"`
 }
 
+type Match struct {
+	Type  MatcherType `json:"type"`
+	Value string      `json:"value"`
+}
+
+type MatcherType string
+
+const (
+	Prefix MatcherType = "Prefix"
+	Regex  MatcherType = "Regex"
```

Comment:
lets call this `RegularExpression` keeping it inline with upstream https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HeaderMatchType